File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/java/cwe-276/code_before1.java
// From cwe-snippets, snippets_2/non-compliant/Java/0138.java

private void WriteToFile(String what_to_write) {
    try{
        File root = Environment.getExternalStorageDirectory();
        if(root.canWrite()) {
            File dir = new File(root + "write_to_the_SDcard");
            File datafile = new File(dir, number + ".extension");
            FileWriter datawriter = new FileWriter(datafile);
            BufferedWriter out = new BufferedWriter(datawriter);
            out.write(what_to_write);
            out.close();
            }
    }
}

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/c/cwe-787/code_before1.c
// From cwe-snippets, snippets_2/non-compliant/C/0005.c 

char *lccopy(const char *str) {
char buf[BUFSIZE];
char *p;
strcpy(buf, str);
for (p = buf; *p; p++) {
  if (isupper(*p)) {
   *p = tolower(*p);
  }
}
return strdup(buf);
}